### PR TITLE
Add storage class diagram

### PR DESCRIPTION
**Pull request by sRanay**

### CODE REVIEW

File: docs/uml/storage.puml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    + validBoolean(booleanStr:String):boolean
+    + convertToGoal(name:String):GOAL
+    + convertToCategory(name:String):Category
+    + prepareTransaction(description:String, amount:double, date:LocalDate, recurrence:String, hasRecurrence:String):Transaction
+    + loadGoal()
+    + loadCategory()
+    + loadIncome()
+    + loadExpense()
+    + load()
+    + saveCategory()
```
Comment (Jonoans) [inline docs/uml/storage.puml:36]: Missing saveGoal()

File: docs/uml/storage.puml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+@startuml
+hide circle
+skinparam classAttributeIconSize 0
+class Storage {
+    - {static} GOAL_STORAGE_FILENAME:String
+    - {static} CATEGORY_STORAGE_FILENAME:String
+    - {static} INCOME_STORAGE_FILENAME:String
+    - {static} EXPENSE_STORAGE_FILENAME:String
+    - {static} FORMATTER:DateTimeFormatter
```
Comment (Jonoans) [inline docs/uml/storage.puml:9]: Missing DATE_PATTERN

File: docs/uml/storage.puml (hunk 1 above)
Comment (sRanay) [reply docs/uml/storage.puml]: Fixed

File: docs/uml/storage.puml (hunk 2 above)
Comment (sRanay) [reply docs/uml/storage.puml]: Fixed

**Review by Jonoans — APPROVED** [review]
LGTM

**Pull request merged**